### PR TITLE
made sure proxy started up with good headers by default

### PR DESCRIPTION
**Pull request by timothy-spencer**

### CODE REVIEW

**Comment by timothy-spencer** [comment]
If it helps, I have tested the new oauth2_proxy code out with `go test` and their `test.sh` script.

File: jobs/oauth2-proxy/templates/bin/ctl [hunk 1]
```diff
@@ -24,6 +24,8 @@ case $1 in
         --client-secret="<%= p('client_secret') %>" \
         --cookie-secret="<%= p('cookie_secret') %>" \
         --email-domain="<%= p('email_domain') %>" \
+        --httpBrowserXssFilter=true \
```
Comment (jmcarp) [inline jobs/oauth2-proxy/templates/bin/ctl]: WDYT about putting these options in the job spec?

Comment (timothy-spencer) [reply jobs/oauth2-proxy/templates/bin/ctl]: Is there any reason why we would want these options ever to be turned off?  Seems like extra yak shaving for something we'll never do.  But if you think this is a good idea, I'm happy to dig in and make that happen.

**Comment by timothy-spencer** [comment]
I think I'll keep the commandline flags the same because it makes them easier to understand in the context of https://github.com/unrolled/secure.

**Comment by sharms** [comment]
The casing in the BOSH release should be consistent - BOSH releases populate templates to execute commands which may have differing casing, but the BOSH release itself and the spec should be in line with itself.

**Comment by jmcarp** [comment]
Looks like this patch updates `.gitmodules` but not the submodule content. I pushed an update to the submodule content as well.

**Comment by timothy-spencer** [comment]
@sharms I'm confused by your review.  Does the spec file need to be renamed?  I didn't change it, I only added the oauth2_proxy stuff to it's contents.

**Comment by timothy-spencer** [comment]
This has been tested out in staging.

**Pull request merged**